webapi/src/main/scala/org/knora/webapi/messages/v2/responder/resourcemessages/ResourceMessagesV2.scala

```diff
@@ -51,12 +52,118 @@ sealed trait ResourcesResponderRequestV2 extends KnoraRequestV2 {
 case class ResourcesGetRequestV2(resourceIris: Seq[IRI], requestingUser: UserADM) extends ResourcesResponderRequestV2
 
 /**
-  * Requests a preview of a resource.
+  * Requests a preview of a resource. A successful response will be a [[ReadResourcesSequenceV2]].
   *
   * @param resourceIris   the Iris of the resources to obtain a preview for.
   * @param requestingUser the user making the request.
   */
-case class ResourcePreviewRequestV2(resourceIris: Seq[IRI], requestingUser: UserADM) extends ResourcesResponderRequestV2
+case class ResourcesPreviewGetRequestV2(resourceIris: Seq[IRI], requestingUser: UserADM) extends ResourcesResponderRequestV2
+
+/**
+  * Requests a resource as TEI/XML. A successful response will be a [[ResourceTEIGetResponseV2]].
+  *
+  * @param resourceIri           the IRI of the resource to be returned in TEI/XML.
+  * @param textProperty          the property representing the text (to be converted to the body of a TEI document).
+  * @param mappingIri            the IRI of the mapping to be used to convert from standoff to TEI/XML, if any. Otherwise the standard mapping is assumed.
+  * @param gravsearchTemplateIri the gravsearch template to query the metadata for the TEI header, if provided.
+  * @param headerXSLTIri         the IRI of the XSL transformation to convert the resource's metadata to the TEI header.
+  * @param requestingUser        the user making the request.
+  */
+case class ResourceTEIGetRequestV2(resourceIri: IRI, textProperty: SmartIri, mappingIri: Option[IRI], gravsearchTemplateIri: Option[IRI], headerXSLTIri: Option[IRI], requestingUser: UserADM) extends ResourcesResponderRequestV2
+
+/**
+  * Represents a Knora resource as TEI/XML.
+  *
+  * @param header the header of the TEI document, if given.
+  * @param body   the body of the TEI document.
+  */
+case class ResourceTEIGetResponseV2(header: TEIHeader, body: TEIBody) {
+
+    def toXML = {
+
+        s"""<?xml version="1.0" encoding="UTF-8"?>
+           |<TEI version="3.3.0" xmlns="http://www.tei-c.org/ns/1.0">
+                ${header.toXML}
+                ${body.toXML}
+           |</TEI>
+        """.stripMargin
+    }
+
+}
+
+/**
+  * Represents information that is going to be contained in the header of a TEI/XML document.
+  *
+  * @param headerInfo the resource representing the header information.
+  * @param headerXSLT XSLT to be applied to the resource's metadata in RDF/XML.
+  *
+  */
+case class TEIHeader(headerInfo: ReadResourceV2, headerXSLT: Option[String], settings: SettingsImpl) {
+
+    def toXML: String = {
+
+        if (headerXSLT.nonEmpty) {
+
+            // TOOD: convert all dates to Gregorian
```

Comment:
Could you do this TODO? What happens to Julian dates now?

Reply:
some recursive shit is needed: https://github.com/dhlab-basel/Knora/pull/862/commits/d434629a523364b2cdcc3d7346321598bddfe3f1